app/src/main/java/com/example/cryptodetails/ui/map/MapsFragment.kt

```diff
@@ -0,0 +1,154 @@
+package com.example.cryptodetails.ui.map
+
+import android.Manifest
+import android.app.AlertDialog
+import android.content.pm.PackageManager
+import android.graphics.Bitmap
+import android.graphics.drawable.Drawable
+import android.location.Address
+import android.location.Geocoder
+import android.net.Uri
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.content.ContextCompat
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.target.CustomTarget
+import com.bumptech.glide.request.transition.Transition
+import com.example.cryptodetails.R
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.location.LocationServices
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+import com.google.android.gms.maps.model.BitmapDescriptor
+import com.google.android.gms.maps.model.BitmapDescriptorFactory
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MarkerOptions
+import java.util.*
+
+
+class MapsFragment : SupportMapFragment() {
+
+    private val onMapReadyCallback = OnMapReadyCallback { googleMap ->
+        googleMap.isMyLocationEnabled = true
+        googleMap.uiSettings.isMyLocationButtonEnabled = true
+        try {
+            LocationServices.getFusedLocationProviderClient(requireActivity()).lastLocation.addOnCompleteListener {
```

Comment:
get current location instead of last location when the user moves, the cursor is not updated with the image.